Analyzer/src/AnalyzeLepTrigger.cc

```diff
@@ -109,356 +100,144 @@ void AnalyzeLepTrigger::Loop(NTupleReader& tr, double weight, int maxevents, boo
-        double weight               = 1.0;
-        double eventweight          = 1.0;
+        double theweight            = 1.0;
         double leptonScaleFactor    = 1.0;
         double bTagScaleFactor      = 1.0;
         double htDerivedScaleFactor = 1.0;
         double topPtScaleFactor     = 1.0;
         double prefiringScaleFactor = 1.0;
         double puScaleFactor        = 1.0;
-        
+
         if(runtype == "MC")
         {
             if( !passMadHT ) continue; //Make sure not to double count DY events
             // Define Lumi weight
             const auto& Weight  = tr.getVar<double>("Weight");
             const auto& lumi = tr.getVar<double>("Lumi");
-            eventweight = lumi*Weight;
-            
+            const auto& puWeight = tr.getVar<double>("puWeightCorr");
+
             // Define lepton weight
             if(NGoodLeptons == 1)
             {
-                const auto& eleLepWeight = tr.getVar<double>("totGoodElectronSF");
-                const auto& muLepWeight  = tr.getVar<double>("totGoodMuonSF");
+                const auto& eleLepWeight = tr.getVar<double>("noTrigGoodElectronSF");
+                const auto& muLepWeight  = tr.getVar<double>("noTrigGoodMuonSF");
                 leptonScaleFactor = (GoodLeptons[0].first == "e") ? eleLepWeight : muLepWeight;
             }
-            
-            //PileupWeight = tr.getVar<double>("_PUweightFactor");
-            bTagScaleFactor   = tr.getVar<double>("bTagSF_EventWeightSimple_Central");
-            htDerivedScaleFactor = tr.getVar<double>("htDerivedweight");
-            topPtScaleFactor = tr.getVar<double>("topPtScaleFactor");
-            prefiringScaleFactor = tr.getVar<double>("prefiringScaleFactor");
-            puScaleFactor = tr.getVar<double>("puWeightCorr");
-            
-            weight *= eventweight*leptonScaleFactor*bTagScaleFactor*htDerivedScaleFactor*prefiringScaleFactor*puScaleFactor;
-        }
-       
-        //Define trigger cuts
-        std::vector<std::string> myMuonTriggers             { "HLT_IsoMu24_v", "HLT_IsoTkMu24_v", "HLT_Mu50_v", "HLT_TkMu50_v" };
-        std::vector<std::string> myElectronTriggers         { "HLT_Ele27_WPTight_Gsf_v", "HLT_Photon175_v", "HLT_Ele115_CaloIdVT_GsfTrkIdT_v" };
 
+            const auto& bTagScaleFactor   = tr.getVar<double>("bTagSF_EventWeightSimple_Central");
+            const auto& htDerivedScaleFactor = tr.getVar<double>("htDerivedweight");
+            const auto& topPtScaleFactor = tr.getVar<double>("topPtScaleFactor");
+            const auto& prefiringScaleFactor = tr.getVar<double>("prefiringScaleFactor");
 
-        if( runYear == "2017" ) {
-            std::vector<std::string> myMuonTriggers         { "HLT_IsoMu24", "HLT_IsoMu27", "HLT_Mu50", "HLT_TkMu50" };
-            std::vector<std::string> myElectronTriggers     { "HLT_Ele35_WPTight_Gsf", "HLT_Photon200", "HLT_Ele115CaloIdVT_GsfTrkIdT" };
+            theweight = lumi*Weight*puWeight;
```

Comment:
Go ahead and use all the scale factors since we know that is the right thing to do.

Reply:
All are multiplied.